src/MonoTorrent/MonoTorrent.Client/Modes/MetadataMode.cs

```diff
@@ -189,7 +189,15 @@ protected override void HandleLtMetadataMessage (PeerId id, LTMetadata message)
                                 try {
                                     if (!Directory.Exists (Path.GetDirectoryName (savePath)))
                                         Directory.CreateDirectory (Path.GetDirectoryName (savePath));
-                                    File.Delete (savePath);
+
+                                    if(true == File.Exists(savePath))
+                                        File.Delete (savePath);
+                                    else//savePath is a directory
+                                    {
+                                        savePath += "\\" + t.Name + ".torrent";
```

Comment:
We should use `Path.Combine` so paths are combined correctly on Windows and Linux/MacOS